ebpf/xdp_filter.c

```diff
@@ -287,6 +309,37 @@ static int __always_inline filter_ipv4(struct xdp_md *ctx, void *data, __u64 nh_
 #endif
     }
 
+#if ENCRYPTED_TLS_BYPASS
+    if ((dport == __constant_ntohs(443)) || (sport == __constant_ntohs(443))) {
```

Comment:
this seems to open a big hole in the ids... as long as you send packets that have port 443 as src or dst (tcp and udp?) and use these 3 first bytes, Suricata will be blind to them?

Reply:
Yes, full bypass it is.

Reply:
I can make it deactivated by default and let the user decide.

Reply:
I don't see how this could ever be a good idea really. I mean you have no idea if the packets are really encrypted and part of a valid TLS stream.

I can see why you'd want it, but this seems to open up things way too much.

Are there other ways we could do this? Track TLS more properly in the filter or re-evaluate the in flight packets, or maybe other things can be done?

Reply:
OK, updated, going to propose an update